File: divisibleBy13CW.ts
function thirt(n: number): number {
    let arr: number[] = [1, 10, 9, 12, 3, 4]
    let newN:number = 0
    while (n > 99){
        let i: number = 0
        while(i < n.toString().length){
            
            newN += (arr[i%arr.length] * parseInt(n.toString().split('').reverse().join('')[i]))
            i += 1
            
        }
        n = newN
        newN = 0
    }
    
    return n
}

File: divisible_by_13_cw.rb
require 'byebug'

def thirt(n)
    arr = [1, 10, 9, 12, 3, 4]
    newN = 0
    while n > 99 do
        i = 0
        while i < n.digits.length do
            newN += (arr[i%arr.length] * n.digits[i]) 
            i += 1
        end
        n = newN
        newN = 0
    end
    return n
end

p thirt(8529)